sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuOverrides.scala

```diff
@@ -1304,6 +1304,29 @@ object GpuOverrides {
         override def convertToGpu(lhs: Expression, rhs: Expression): GpuExpression =
           GpuTimeAdd(lhs, rhs)
       }),
+    expr[DateAddInterval](
+      "Adds interval to date",
+      ExprChecks.binaryProjectNotLambda(TypeSig.DATE, TypeSig.DATE,
+        ("start", TypeSig.DATE, TypeSig.DATE),
+        ("interval", TypeSig.lit(TypeEnum.CALENDAR)
+          .withPsNote(TypeEnum.CALENDAR, "month intervals are not supported"),
+          TypeSig.CALENDAR)),
+      (a, conf, p, r) => new BinaryExprMeta[DateAddInterval](a, conf, p, r) {
+        override def tagExprForGpu(): Unit = {
+          GpuOverrides.extractLit(a.interval).foreach { lit =>
+            val intvl = lit.value.asInstanceOf[CalendarInterval]
+            if (intvl.months != 0) {
+              willNotWorkOnGpu("interval months isn't supported")
+            }
+          }
+          if (ZoneId.of(a.timeZoneId.get).normalized() != GpuOverrides.UTC_TIMEZONE_ID) {
```

Comment:
I think it will be simpler to read 

```
if (a.timeZoneId.nonEmpty && ZoneId.of(a.timeZoneId.get).normalized() != GpuOverrides.UTC_TIMEZONE_ID) {
...
}
```
I am also not sure what the performance impact will be but its not a big deal as its not huge

Reply:
Thanks @razajafri . I have incorporated @gerashegalov suggestion here. Let me know if it's okay.

Reply:
Can you also fix it in TimeAdd and other places where we might be doing this?

Reply:
Sure, Do you want me to fix it in this PR or as a follow on ? I can put up another PR with just those fixes for other expressions.